util/patch/patch.go

```diff
@@ -124,11 +130,21 @@ func (h *Helper) Patch(ctx context.Context, obj client.Object, opts ...Option) e
 	}
 
 	// Calculate and store the top-level field changes (e.g. "metadata", "spec", "status") we have before/after.
-	h.changes, err = h.calculateChanges(obj)
+	patchDiff, err := h.calculateChanges(obj)
 	if err != nil {
 		return err
 	}
 
+	// log whether a patch is created. If enabled log the full patch.
+	if len(patchDiff) == 0 {
+		log.Info(fmt.Sprintf("No changes for %s", tlog.KObj{Obj: h.after}))
```

Comment:
Q: Is it possible to use klog.KObj given that this is what we aim for?

Reply:
It is possible, just the question if we want to do it already

Reply:
I think we should leave this for now and try to do a replace across our codebase as a cleanup once we're sure of the format in the upstream library.

Reply:
The format is pretty safe. just `klog.KObj` as value and lower case singlar kind as key. 
I'm fine with both optoins.